pkg/kubelet/cpumanager/cpu_manager.go

```diff
@@ -0,0 +1,363 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpumanager
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/record"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+	"k8s.io/kubernetes/pkg/kubelet/cpumanager/state"
+	"k8s.io/kubernetes/pkg/kubelet/cpumanager/topology"
+	"k8s.io/kubernetes/pkg/kubelet/eviction"
+	"k8s.io/kubernetes/pkg/kubelet/lifecycle"
+	"k8s.io/kubernetes/pkg/kubelet/status"
+	"k8s.io/kubernetes/pkg/kubelet/util/format"
+)
+
+type kletGetter interface {
+	GetPods() []*v1.Pod
+	GetCachedMachineInfo() (*cadvisorapi.MachineInfo, error)
+}
+
+type runtimeService interface {
+	UpdateContainerResources(id string, resources *runtimeapi.LinuxContainerResources) error
+}
+
+type policyName string
+
+// Manager interface provides methods for kubelet to manage pod cpus
+type Manager interface {
+	Start()
+
+	// RegisterContainer registers a container with the cpuset manager
+	// resulting in synchronous setting on the container cpuset.  This
+	// is called after RegisterContainer(), which creates the containerID, and
+	// before StartContainer(), such that the cpuset is configured before
+	// the container starts
+	RegisterContainer(p *v1.Pod, c *v1.Container, containerID string) error
```

Comment:
How about:

```
// RegisterContainer is called between container creation and container start
// so that initial CPU affinity settings can be written through to the
// container runtime before the first process begins to execute.
```